core/src/main/java/apoc/trigger/TriggerHandlerRead.java

```diff
@@ -0,0 +1,228 @@
+package apoc.trigger;
+
+import apoc.Pools;
+import apoc.SystemLabels;
+import apoc.SystemPropertyKeys;
+import apoc.util.Util;
+import org.neo4j.dbms.api.DatabaseManagementService;
+import org.neo4j.graphdb.GraphDatabaseService;
+import org.neo4j.graphdb.Node;
+import org.neo4j.graphdb.Result;
+import org.neo4j.graphdb.Transaction;
+import org.neo4j.graphdb.event.TransactionData;
+import org.neo4j.graphdb.event.TransactionEventListener;
+import org.neo4j.internal.helpers.collection.Iterators;
+import org.neo4j.internal.helpers.collection.MapUtil;
+import org.neo4j.kernel.lifecycle.LifecycleAdapter;
+import org.neo4j.logging.Log;
+import org.neo4j.scheduler.Group;
+import org.neo4j.scheduler.JobHandle;
+import org.neo4j.scheduler.JobScheduler;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static apoc.ApocConfig.apocConfig;
+import static apoc.trigger.TriggerHandlerWrite.getTriggerNodes;
+import static apoc.trigger.TriggerHandlerWrite.withSystemDb;
+
+
+public class TriggerHandlerRead extends LifecycleAdapter implements TransactionEventListener<Void> {
+
+    public static final int TRIGGER_RELOAD_DEFAULT_VALUE = 3000;
+    public static final String TRIGGER_RELOAD = "apoc.trigger.reload";
+
+    private enum Phase {before, after, rollback, afterAsync}
+    
+    private final ConcurrentHashMap<String, Map<String,Object>> activeTriggers = new ConcurrentHashMap<>();
+
+    private final Log log;
+    private final GraphDatabaseService db;
+    private final DatabaseManagementService databaseManagementService;
+    private final Pools pools;
+    
+    private final JobScheduler jobScheduler;
+
+    private long lastUpdate;
+
+    private JobHandle restoreTriggerHandler;
+
+    private final AtomicBoolean registeredWithKernel = new AtomicBoolean(false);
+
+    public TriggerHandlerRead(GraphDatabaseService db, DatabaseManagementService databaseManagementService,
+                              Log log, Pools pools, JobScheduler jobScheduler) {
+        this.db = db;
+        this.databaseManagementService = databaseManagementService;
+        this.log = log;
+        this.pools = pools;
+        this.jobScheduler = jobScheduler;
+    }
+
+
+    public void reconcileKernelRegistration() {
+        final boolean hasTriggers = withSystemDb(tx -> getTriggerNodes(db.databaseName(), tx)
+                .hasNext());
+        reconcileKernelRegistration(hasTriggers);
+    }
```

Comment:
If we agree that reconcileKernelRegistration should be a private method and should not be called from the controller class (see my other comment below) then the only place this is called is from updateCache().

If the only place this called from is updateCache() then this method should not be needed, and will only make things slower. This is because updateCache() can directly call reconcileKernelRegistration(true | false) without having to retrieve the trigger nodes again from the system database.